### PR TITLE
fix: default to electron for runner-ct

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://cypress-io.atlassian.net/browse/CT-403

### User facing changelog

By default Component Testing will launch in Electron. It looks nicer and is a more consistent experience. Also, this change means users do not need to install a browser on CI - the tests will just run in the electron browser that comes with Cypress.

image

### Additional details

We should default to using the bundled electron browser by default. This is so when you run CT on CI, users do not need to install a browser. Instead, it will just use the Electron that is bundled with Cypress.

### How has the user experience changed?

- Launch Electron by default in open-ct
- Headless by default in run-ct

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by JessicaSachs** [comment]
There's an issue with this PR in which it orphans the renderer process when Cypress is closed. This leads to one orphaned ~100MB process per Cypress application close. This issue does not exist in Chrome. We can either:
1. Change the default to chrome and ship with the orphaned process
2. Fix the child process from being orphaned

Going to investigate #2 for 30mins to see if it's an easy fix. If not, we'll default to Chrome and ship.

**Review by JessicaSachs — CHANGES_REQUESTED** [review]
Blocked pending the fix for orphaning child processes

**Comment by brian-mann** [comment]
I'm looking at this now

**Comment by JessicaSachs** [comment]
This is worse that initially thought. The Electron application crashes and the error code isn't handled properly. This means that if tests fail, we exit with exit code 0 and pass the tests. Brian and Ben and I are looking into this, but we can't ship without fixing it.

File: packages/server/lib/modes/run.js [hunk 1]
```diff
@@ -1161,7 +1161,7 @@ module.exports = {
     // to avoid chopping off the end of the video
     const delay = this.getVideoRecordingDelay(startedVideoCapture)
 
-    return this.listenForProjectEnd(project, exit, runAllSpecsInSameBrowserSession)
```
Comment (lmiller1990) [inline packages/server/lib/modes/run.js:1164]: unused

**Comment by JessicaSachs** [comment]
Update: This PR was attempting to add support for Electron, but the crashers in this PR were masked by an unrelated bug that is now fixed by #15727. After isolating the issue, I was able to fix the root cause of the Electron-only failure. More details are available in #15731 which replaces this PR.

**Pull request closed without merging**